src/utils/chartJs/CustomTooltip.jsx

```diff
@@ -1,57 +1,85 @@
 import React from 'react';
 import { withStyles } from '@material-ui/core/styles';
 
+const topAligned = {
+  '--trans-y': 'var(--tip-size)',
+  '&::before': { bottom: '100%', borderBottomColor: 'var(--bg-color)' },
+  '&.xright': {
+    '--trans-x': 'calc(-100% + 2 * var(--tip-size))',
+    '&::before': { right: 'var(--tip-size)' },
+  },
+  '&.xcenter': {
+    '--trans-x': '-50%',
+    '&::before': { left: 'calc(50% - var(--tip-size))' },
+  },
+  '&.xleft': {
+    '--trans-x': 'calc(-2 * var(--tip-size))',
+    '&::before': { left: 'var(--tip-size)' },
+  },
+};
```

Comment:
For 'top' and 'bottom' alignment, the custom position style share a lot of similarities (except two lines). The only way I could think of to share the same code in both cases was to define one of them as constant and share it.

Check (if you wish) my [raw diagrams](https://docs.google.com/presentation/d/1i-Xz7z6NqzQwiWYsgtjvrZP-t6JAbzfMfA1K0b1eG7w/edit?usp=sharing) to see the alignments.